nixos/tests/containers-extra_veth.nix

```diff
@@ -52,52 +52,59 @@ import ./make-test.nix ({ pkgs, ...} : {
 
   testScript =
     ''
-      $machine->waitForUnit("default.target");
-      $machine->succeed("nixos-container list") =~ /webserver/ or die;
+      machine.wait_for_unit("default.target")
+      out = machine.succeed("nixos-container list")
+      assert "webserver" in out
 
-      # Status of the webserver container.
-      $machine->succeed("nixos-container status webserver") =~ /up/ or die;
+      with subtest("Status of the webserver container"):
+          out = machine.succeed("nixos-container status webserver")
+          assert "up" in out
 
       # Debug
-      #$machine->succeed("nixos-container run webserver -- ip link >&2");
+      # machine.succeed("nixos-container run webserver -- ip link >&2")
 
-      # Ensure that the veths are inside the container
-      $machine->succeed("nixos-container run webserver -- ip link show veth1") =~ /state UP/ or die;
-      $machine->succeed("nixos-container run webserver -- ip link show veth2") =~ /state UP/ or die;
+      with subtest("Ensure that the veths are inside the container"):
+          out = machine.succeed("nixos-container run webserver -- ip link show veth1")
+          assert "state UP" in out
+          out = machine.succeed("nixos-container run webserver -- ip link show veth2")
+          assert "state UP" in out
 
       # Debug
-      #$machine->succeed("ip link >&2");
+      # machine.succeed("ip link >&2")
 
-      # Ensure the presence of the extra veths
-      $machine->succeed("ip link show veth1") =~ /state UP/ or die;
-      $machine->succeed("ip link show veth2") =~ /state UP/ or die;
+      with subtest("Ensure the presence of the extra veths"):
+          out = machine.succeed("ip link show veth1")
+          assert "state UP" in out
+          out = machine.succeed("ip link show veth2")
+          assert "state UP" in out
 
-      # Ensure the veth1 is part of br1 on the host
-      $machine->succeed("ip link show veth1") =~ /master br1/ or die;
+      with subtest("Ensure the veth1 is part of br1 on the host"):
+          out = machine.succeed("ip link show veth1")
+          assert "master br1" in out
 
       # Debug
```

Comment:
there's a lot to comment out in debug cases. Is it likely that all these prints will help in case this test fails in the future? if so, why not add some constant `DEBUG_OUTPUT` in the beginning of the test and then put these commands into some conditional - then the debugging user can simply set one variable to `True` and get it all with less work.